databricks/koalas/series.py

```diff
@@ -1907,6 +1908,136 @@ def _fillna(self, value=None, method=None, axis=None, limit=None, part_cols=()):
+        0    1
+        1   -2
+        2    0
+        3    2
+        4    1
+        dtype: int64
+        >>> uniques
+        Index(['a', 'b', 'c'], dtype='object')
+        """
+        assert (na_sentinel is None) or isinstance(na_sentinel, int)
+        assert sort is True
+        uniq_sdf = self._internal.spark_frame.select(self.spark.column).distinct()
+
+        # Check number of uniques and constructs sorted `uniques_list`
+        max_compute_count = get_option("compute.max_rows")
+        if max_compute_count is not None:
+            uniq_pdf = uniq_sdf.limit(max_compute_count + 1).toPandas()
+            if len(uniq_pdf) > max_compute_count:
+                raise ValueError(
+                    "Current Series has more then {0} unique values. "
+                    "Please set 'compute.max_rows' by using 'databricks.koalas.config.set_option' "
+                    "to more than {0} rows. Note that, before changing the "
+                    "'compute.max_rows', this operation is considerably expensive.".format(
+                        max_compute_count
+                    )
+                )
+        else:
+            raise ValueError(
+                "Please set 'compute.max_rows' by using 'databricks.koalas.config.set_option' "
+                "to restrict the total number of unique values of the current Series."
+                "Note that, before changing the 'compute.max_rows', "
+                "this operation is considerably expensive."
+            )
+
+        uniques_list = first_series(uniq_pdf).tolist()
+        uniques_list = sorted(uniques_list, key=lambda x: (x is None, x))
+
+        # Constructs `unique_to_code` mapping unique to code
+        unique_to_code = {}
+        if na_sentinel is not None:
+            unique_to_code = {np.nan: na_sentinel}
+            na_sentinel_code = na_sentinel
+        code = 0
+        for unique in uniques_list:
+            if unique not in unique_to_code:
+                if unique is None and na_sentinel is not None:
+                    continue
+
+                if unique is None:
+                    na_sentinel_code = code
+                else:
+                    unique_to_code[unique] = code
+                code += 1
+
+        sdf = self._internal.resolved_copy.spark_frame
+        scol_name = self._internal.data_spark_column_names[0]
+        new_scol_name = cast(str, verify_temp_column_name(sdf, "__new_column__"))
+
+        kvs = list(
+            chain(*([(F.lit(unique), F.lit(code)) for unique, code in unique_to_code.items()]))
+        )
+        sdf = sdf.withColumn(new_scol_name, F.create_map(kvs))
+
+        null_scol = F.when(scol_for(sdf, scol_name).isNull(), F.lit(na_sentinel_code))
+        mapped_scol = scol_for(sdf, new_scol_name).getItem(scol_for(sdf, scol_name))
+        sdf = sdf.withColumn(scol_name, null_scol.otherwise(mapped_scol))
+
+        codes = first_series(DataFrame(self._internal.with_new_sdf(sdf))).astype(np.int64)
```

Comment:
I guess we can make it not create a new Spark DataFrame for `codes`.

Reply:
Thank you! Updated.